### PR TITLE
Modal pages

### LINKED ISSUE #166: [FE] In the redeem section, add modal with the status of the transaction
In the redeem section, add modal with the status of the transaction
Also, the modal should be added to the
- Report payouts page
- Merge positions page

The modal should contain OK button, which closes the pop-up and leaves a user on the page he/she is currently on.

### LINKED ISSUE #167: [FE] Change the status of the modal in every submit button to success if the transaction is OK
Related to the next pages: 
1. Split position
2. Report payouts page
3. Merge positions page
4. Redeem position page

### LINKED ISSUE #296: Report Payout page: validation message about a transaction rejection appears when try co confirm the transaction once again
**Steps:** 
1. Open Report Payout page
2. Fill in all the required fields with valid data
3. Press on the Report button --> Reject in Metamask (or any connected app)
4. Do not reload the page, press on the Report button --> Confirm in Metamask

**AR:** validation message about a transaction rejection appears
https://drive.google.com/file/d/1ftyHul_yrE6AbxLo2k9CJV7w4K3WQxgo/view

**ER:** validation message about a transaction rejection should appear when reject the aciton. The user should be able to comlete the transaction, no validation messages should be shown

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #166 
Closes #167 
Closes #296 

It looks like this
Captura de pantalla de 2020-09-22 12-33-38

I've added the logic for checking if is it a partial merge and also tests many possible situations.
- If you have splitted N positions from a collateral and merge them, you get the collateral back.
- If you have spllited N positions from a collateral and merge n with n < N, then you get another position of the same depth. In this case, you can use all your 'position collateral' or maintain some balance of the original positions 
- If you have spllited N positions from another position, is the same logic with the only difference that you get balance of your original position when you merge the full index set.

So, for testing what I did is  (everything is in rinkeby):

- A condition with 4 outcomes, `0x422f8c3a2e4a805e548e3fd33de64fa636bbfd70acd48128eeb015ddafbd26bf` split into 4 positions [ `0x355e079b7da5a31d128caa0801d9f58e8edd1e590e49e027f303ee2cf366af13`, `0xce08f2b81436386733315e04b85e3b80d47e31d4c07c1762f6ee440341f45c0f`, `0xe92095e0d6fdebe03adcb9e7f6775f7bf118702428ed380d8b8338889b2a1f02`, `0x775006f64785fc4881f4a69b7977f8865a1a4197b62072a9a3b07cb92e589b6c` ]. 
I merged two of them (second and four, index sets: 2 and 8) by an amount of 8, and I get ` 0x106e28ec6f3a23d891804f8895d309b66d39cc03ba5f1540bc158911050d2377`. 
- With this I can merge the new position with the other two to complete a full index set and obtain 8 DAI back. And maintain positions with balance = 2 (the rest that was not merged).

### CODE REVIEW

File: src/pages/MergePositions/Contents.tsx [hunk 1]
```diff
@@ -131,10 +135,14 @@ export const Contents = () => {
...
-        clearCondition()
-        updateBalances([])
+        //if freeindexset == 0, everything is merged
+        if (parentCollectionId === NULL_PARENT_ID) {
+          setMergeResult(collateralToken.id)
+        } else {
+          setMergeResult(
+            ConditionalTokensService.getPositionId(collateralToken.id, parentCollectionId)
+          )
+        }
```
Comment (hpmaxi) [inline src/pages/MergePositions/Contents.tsx:145]: This should check for partial merges.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr306--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr306--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — CHANGES_REQUESTED** [review]
For "Report Payouts" you are showing this (a modal) when finished:

Screen Shot 2020-09-22 at 15 20 56

But it should look like this ("FullLoading" component, not a modal, you should obviously change the wording), you can grab some code from _"Transfer Outcome Tokens":_

Screen Shot 2020-09-22 at 15 28 35

----

Same here ("Redeem Positions"):

Screen Shot 2020-09-22 at 15 36 21

----

For "Merge Positions" you are showing a modal like this:

Screen Shot 2020-09-22 at 15 45 42

... which is fine, but you should be showing more info, like what we have for "Split Position"'s result:

Screen Shot 2020-09-22 at 15 50 56

Keep in mind that you should use a "FullLoading" component in case there's an error though.

----

Regarding the "FullLoading" components, generally speaking the format should be:

- **Title:** Merge Positions / Report Payouts / Redeem Positions
- **Icon:** Whatever the status is (Error / Success).
- **Text:** "Working..." if working / "All done!" or "Finished!" when finished / "There was an error." or the full error string if available.
- Green "OK" button for success, red "Close" button for errors.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr306--conditionaltokensexplorer.review.gnosisdev.com

**Comment by hpmaxi** [comment]
I've updated the original description and it's working for partial merging too.

**Review by mariano-aguero — APPROVED** [review]
There are a lot of details that we can work in another issue, @gabitoesmiapodo will create it

**Review by gabitoesmiapodo — APPROVED** [review]
Good enough, but it needs a little work to achieve greatness.

I will create a few issues so we can achieve it.

**Pull request merged**